src/libraries/System.Private.CoreLib/src/System/DefaultBinder.cs

```diff
@@ -383,8 +383,10 @@ internal partial class DefaultBinder : Binder
 #endregion
             }
 
+            MethodBase bestMatch = candidates[currentMin]!;
```

Comment:
Hoisted up to use in the throw (note the trailing `!` that was in all the references below. We KNOW it's no-null.